soccerdata/fotmob.py

```diff
@@ -0,0 +1,434 @@
+import itertools
+import json
+from pathlib import Path
+from typing import Callable, Dict, Iterable, List, Optional, Union
+
+import pandas as pd
+
+from ._common import BaseRequestsReader, make_game_id, season_code
+from ._config import DATA_DIR, NOCACHE, NOSTORE, TEAMNAME_REPLACEMENTS, logger
+
+FOTMOB_DATADIR = DATA_DIR / 'Fotmob'
+FOTMOB_API = 'https://www.fotmob.com/api/'
+
+
+class Fotmob(BaseRequestsReader):
+    """Provides pd.DataFrames from data available at http://www.fotmob.com.
+
+    Data will be downloaded as necessary and cached locally in
+    ``~/soccerdata/data/Fotmob``.
+
+    Parameters
+    ----------
+    leagues : string or iterable, optional
+        IDs of Leagues to include.
+    seasons : string, int or list, optional
+        Seasons to include. Supports multiple formats.
+        Examples: '16-17'; 2016; '2016-17'; [14, 15, 16]
+    proxy : 'tor' or dict or list(dict) or callable, optional
+        Use a proxy to hide your IP address. Valid options are:
+            - 'tor': Uses the Tor network. Tor should be running in
+              the background on port 9050.
+            - dict: A dictionary with the proxy to use. The dict should be
+              a mapping of supported protocols to proxy addresses. For example::
+
+                  {
+                      'http': 'http://10.10.1.10:3128',
+                      'https': 'http://10.10.1.10:1080',
+                  }
+
+            - list(dict): A list of proxies to choose from. A different proxy will
+              be selected from this list after failed requests, allowing rotating
+              proxies.
+            - callable: A function that returns a valid proxy. This function will
+              be called after failed requests, allowing rotating proxies.
+    no_cache : bool
+        If True, will not use cached data.
+    no_store : bool
+        If True, will not store downloaded data.
+    data_dir : Path
+        Path to directory where data will be cached.
+    path_to_browser : Path, optional
+        Path to the Chrome executable.
+    headless : bool, default: True
+        If True, will run Chrome in headless mode. Setting this to False might
+        help to avoid getting blocked.
+    """
+
+    def __init__(
+        self,
+        leagues: Optional[Union[str, List[str]]] = None,
+        seasons: Optional[Union[str, int, Iterable[Union[str, int]]]] = None,
+        proxy: Optional[
+            Union[str, Dict[str, str], List[Dict[str, str]], Callable[[], Dict[str, str]]]
+        ] = None,
+        no_cache: bool = NOCACHE,
+        no_store: bool = NOSTORE,
+        data_dir: Path = FOTMOB_DATADIR,
+    ):
+        """Initialize the Fotmob reader."""
+        super().__init__(
+            leagues=leagues,
+            proxy=proxy,
+            no_cache=no_cache,
+            no_store=no_store,
+            data_dir=data_dir,
+        )
+        self.seasons = seasons  # type: ignore
+        self.rate_limit = 3
```

Comment:
This is a very high rate limit. Does FotMob implement some kind of countermeasure against scraping?

Reply:
Not that I'm aware of; I just cautiously set the same rate limit to be the same as fbref, but can see what happens when removed.